lib/multi-level.js

```diff
@@ -120,13 +119,16 @@ class MultiLevelCache {
           expiry: Date.now() + this._maxAge
         };
 
-        // Out-of-band cache update
-        // NB: an in-band update would `await` this Promise.all block
-        Promise.all(this._caches.map(cache => cache.set(key, cacheItem))).catch(err => {
-          throw new Error(`Error caching ${key}`, err);
-        }).then(() => {
-          delete this._pendingRefreshes[key];
-        });
+        // Given that we are not ignoring this value, perform an out-of-band cache update
+        const willCache = shouldCache || this.shouldCache;
+        if (willCache(cacheItem.value)) {
```

Comment:
Maybe do a type check to make sure it's actually a function before you call it?

Reply:
`shouldCache` and `this.shouldCache` are already defined in the `JSDoc` to be of the following type:

```js
/**
 * @callback ShouldCache
 * @param {GetResult} a cache item that we are electing to cache (or not)
 * @returns {Boolean}
 */
```

But I can add a `typeof` check anyway

Reply:
Done in https://github.com/godaddy/out-of-band-cache/pull/2/commits/697c1d39906c9804b40526d2f8b9cdf40882dcc3